phoenix-core/src/main/java/org/apache/phoenix/util/MetaDataUtil.java

```diff
@@ -941,6 +941,16 @@ public static IndexType getIndexType(List<Mutation> tableMetaData, KeyValueBuild
         return null;
     }
 
+    public static PDataType<?> getIndexDataType(List<Mutation> tableMetaData,
+            KeyValueBuilder builder, ImmutableBytesWritable value) {
+        if (getMutationValue(getPutOnlyTableHeaderRow(tableMetaData),
```

Comment:
nit: comment would be really nice here. I _think_ you're looking up the view index id config setting from syscat, and that makes sense, but the code's a mouthful without explanation :-)

Reply:
Done.